pageserver/src/layered_repository.rs

```diff
@@ -1139,8 +1155,11 @@ impl LayeredTimeline {
     fn checkpoint_internal(&self, force: bool) -> Result<()> {
         // FIXME: these should be fetched atomically.
```

Comment:
FIXME comment can be removed now.

Hmm, we probably should be reading last/prev_record_lsn first, and last_valid_lsn only after that. In general, `self.last_valid_lsn` can be ahead of `self.last_record_lsn`, but not the other way round. I'm not sure if anything bad would happen if they are the other way round in these local variables, but it feels wrong.